src/DssGateSuck.sol

```diff
@@ -0,0 +1,166 @@
+
+ MISCELLANEOUS
+  * does not execute vow.heal to ensure the dai draw amount from vat.suck is lower than the surplus buffer currently held in vow
+  * vat, and vow addresses cannot be updated after deployment
+*/
+contract DssGateSuck {
+    // --- Auth ---
+    mapping (address => uint256) public wards;                                       // Addresses with admin authority
+    event Rely(address indexed usr);
+    event Deny(address indexed usr);
+    function rely(address _usr) external auth { wards[_usr] = 1; emit Rely(_usr); }  // Add admin
+    function deny(address _usr) external auth { wards[_usr] = 0; emit Deny(_usr); }  // Remove admin
+    modifier auth {
+        require(wards[msg.sender] == 1, "DssGateSuck/not-authorized");
+        _;
+    }
+
+    // --- Integration Access Control ---
+    mapping (address => uint256) public can;
+    event Hope(address usr);
+    event Nope(address usr);
+    function hope(address _a) external auth {
+        require(_a != address(0), "DssGateSuck/cannot-hope-nothing");
+        can[_a] = 1;
+        emit Hope(_a);
+    }
+    function nope(address _a) external auth {
+        can[_a] = 0;
+        emit Nope(_a);
+    }
+    modifier wish { require(can[msg.sender] == 1, "DssGateSuck/bud-not-authorized"); _; }
+
+    /// maker protocol vat
+    address public immutable vat;
+    /// maker protocol vow
+    address public immutable vow;
+
+    /// draw limit- total amount that can be drawn from vat.suck
+    uint256 public max; // [rad]
+    // amount drawn- amount currently drawn and not put back
+    uint256 public fill;
+
+    constructor(address vat_, address vow_) {
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+
+        vat = vat_; // set vat address
+        vow = vow_; // set vow address
+    }
+
+    // --- Events ---
+    event File(bytes32 indexed what, uint256 data);
+    event Draw(address indexed dst, uint256 amount); // log upon draw
+
+    // --- UTILS ---
+    function _max(uint x, uint y) internal pure returns (uint z) {
+        return x <= y ? y : x;
+    }
+
+    // --- Draw Limits ---
+    /// Update draw limit
+    /// @dev Restricted to authorized governance addresses
+    /// @dev Approved total can be updated to both a higher or lower value
+    /// @param what what value are we updating
+    /// @param data what are we updating it to
+    function file(bytes32 what, uint256 data) external auth {
+        if (what == "max") {
+            max = data; // update approved total amount
+            emit File(what, data);
+        } else revert("DssGateSuck/file-not-recognized");
+    }
+
+    /// Suck to destination
+    /// @dev Restricted to approved integration addresses
+    /// @param dst who are you sucking to
+    /// @param amt dai amount in rad
+    function suck(address dst, uint256 amt) public wish {
+        require(VatLike(vat).live() == 1, "DssGateSuck/vat-not-live");
+
+        fill = fill + amt;
```

Comment:
Need to check if can save a storage read here.
Suggest to try to compare to:
`uint256 _fill = fill + amt;
require(max >= _fill, ...)
fill = _fill`